### DISCUSSION TITLE
Trace context propagation to CompletableFuture and @Async methods

### DISCUSSION THREAD

**Discussion by birkoni5** [open-telemetry/opentelemetry-java-instrumentation#7657] [Q&A] [answered]
Hello,

Is there a way to propagate trace context to `CompletableFuture`  and `@Async` methods?
I'm using the OpenTelemetry Spring Boot Starter.

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @birkoni5 ,

If you wrap the `Executor` you're using for async execution with [`Context#taskWrapping()`](https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/Context.java#L119) the context will automatically propagate to all tasks that are executed by it.

```java
  @Bean
  public Executor taskExecutor() {
    Executor executor = // ... 
    return Context.taskWrapping(executor);
  }
```
